MiniApp/Classes/Storage/MiniAppKeyChain.swift

```diff
@@ -0,0 +1,126 @@
+import Foundation
+
+@objc class MiniAppKeyChain: NSObject {
+    let service: String
+    var account: String
+
+    typealias KeysDictionary = [String: [MASDKCustomPermissionModel]]
+
+    init(service: String = Bundle.main.bundleIdentifier!) {
+        self.service = service
+        self.account = "\(service).rakuten.tech.permission.keys"
+    }
+
+    func getCustomPermissions(forMiniApp keyId: String) -> [MASDKCustomPermissionModel] {
+        guard let allKeys = keys() else {
+            return getDefaultSupportedPermissions()
+        }
+        guard let permissionList = allKeys[keyId] as [MASDKCustomPermissionModel]? else {
+            return setDefaultPermissionsInKeyChain(forMiniApp: keyId)
+        }
+        return permissionList
+    }
+
+    func storeCustomPermissions(permissions: [MASDKCustomPermissionModel], forMiniApp keyId: String) {
+        var keysDic = keys()
+        var cachedPermissions = self.getCustomPermissions(forMiniApp: keyId)
+        _ = permissions.map { (permissionModel: MASDKCustomPermissionModel) -> MASDKCustomPermissionModel in
```

Comment:
did you need to assign it to the `_` ?

Reply:
Yeah, I just want to map permissions and make changes in cachedPermissions. So the result is not needed